File: app/src/main/java/com/omvp/app/ui/samples/requestphone/view/SampleRequestPhoneFragment.java
package com.omvp.app.ui.samples.requestphone.view;

import android.os.Bundle;
import android.support.v7.widget.AppCompatTextView;
import android.view.View;
import android.widget.RelativeLayout;

import com.omvp.app.R;
import com.omvp.app.base.mvp.view.BaseViewFragment;
import com.omvp.app.base.mvp.view.BaseViewFragmentCallback;
import com.omvp.app.ui.samples.requestphone.presenter.SampleRequestPhonePresenterImpl;
import com.omvp.components.InputLayoutView;

import butterknife.BindView;
import butterknife.OnClick;

public class SampleRequestPhoneFragment extends BaseViewFragment<SampleRequestPhonePresenterImpl, SampleRequestPhoneFragment.FragmentCallback>
        implements SampleRequestPhoneView {

    @BindView(R.id.input_prefix)
    InputLayoutView mInputPrefixView;
    @BindView(R.id.input_phone)
    InputLayoutView mInputPhoneView;
    @BindView(R.id.code_received_layout)
    RelativeLayout mCodeReceivedLayout;
    @BindView (R.id.input_code)
    InputLayoutView mInputCodeToSent;
    @BindView(R.id.send)
    AppCompatTextView mButtonGetSent;
    @BindView(R.id.sign_out)
    AppCompatTextView mButtonSignOut;

    public interface FragmentCallback extends BaseViewFragmentCallback {
        void signOut();
    }

    public static SampleRequestPhoneFragment newInstance(Bundle bundle) {
        SampleRequestPhoneFragment fragment = new SampleRequestPhoneFragment();
        bundle = bundle == null ? new Bundle() : bundle;
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public void onViewStateRestored(Bundle savedInstanceState) {
        super.onViewStateRestored(savedInstanceState);

        setupViews();
    }

    private void setupViews() {

    }

    @OnClick(R.id.send)
    public void onValidatePhoneClick(View view) {
        mPresenter.validatePhone(mInputPrefixView.getText(), mInputPhoneView.getText());
    }

    @OnClick(R.id.sign_out)
    public void onSignOutClick(View view) {
       if (mCallback != null) {
           mCallback.signOut ();
       }
    }

    @Override
    public void showPrefixInputError(String error) {
        mInputPrefixView.setError(error);
    }

    @Override
    public void showPhoneInputError(String error) {
        mInputPhoneView.setError(error);
    }

    @Override
    public void showPrefixInputSuccess() {
        mInputPrefixView.setSuccess();
    }

    @Override
    public void showPhoneInputSuccess() {
        mInputPhoneView.setSuccess();
    }

    @Override
    public void drawInputCode(String code) {
        mInputCodeToSent.setText(code);
    }

    @Override
    public void codeReceivedLayoutVisibility(int visibility) {
        mCodeReceivedLayout.setVisibility(visibility);
    }

    public void changeButtonText (String title) {
        mButtonGetSent.setText(title);
    }

    public void singOutButtonVisibility (int visibility) {
        mButtonSignOut.setVisibility(visibility);
    }

}

File: app/src/main/java/com/omvp/app/ui/samples/requestphone/view/SampleRequestPhoneFragment.kt
package com.omvp.app.ui.samples.requestphone.view

import android.os.Bundle
import com.omvp.app.base.mvp.view.BaseViewFragment
import com.omvp.app.base.mvp.view.BaseViewFragmentCallback
import com.omvp.app.ui.samples.requestphone.presenter.SampleRequestPhonePresenterImpl
import kotlinx.android.synthetic.main.sample_request_phone_fragment.*

class SampleRequestPhoneFragment : BaseViewFragment<SampleRequestPhonePresenterImpl, SampleRequestPhoneFragment.FragmentCallback>(), SampleRequestPhoneView {

    interface FragmentCallback : BaseViewFragmentCallback {
        fun signOut()
    }

    companion object {
        fun newInstance(bundle: Bundle?) = SampleRequestPhoneFragment().apply {
            arguments = bundle ?: Bundle()
        }
    }

    override fun onViewStateRestored(savedInstanceState: Bundle?) {
        super.onViewStateRestored(savedInstanceState)
        sign_out.setOnClickListener { onSignOutClick() }
    }

    override fun showPrefixInputError(error: String) {
        input_prefix.setError(error)
    }

    override fun showPhoneInputError(error: String) {
        input_phone.setError(error)
    }

    override fun showPrefixInputSuccess() {
        input_prefix.setSuccess()
    }

    override fun showPhoneInputSuccess() {
        input_phone.setSuccess()
    }

    override fun drawInputCode(code: String) {
        input_code.text = code
    }

    override fun codeReceivedLayoutVisibility(visibility: Int) {
        code_received_layout.visibility = visibility
    }

    fun changeButtonText(title: String) {
        send.text = title
    }

    fun singOutButtonVisibility(visibility: Int) {
        sign_out.visibility = visibility
    }

    private fun onValidatePhoneClick() {
        mPresenter.validatePhone(input_prefix.text, input_phone.text)
    }

    private fun onSignOutClick() {
        mCallback.signOut()
    }
}
